### PR TITLE
Implement API key management

### LINKED ISSUE #4: Store API key securely
See https://developer.android.com/training/articles/keystore.html

### LINKED ISSUE #6: Validate API key

### PR DESCRIPTION
**Pull request by NahuLD**
Resolves #4, resolves #6.

This branch will only add the basic feature of storage and validation for the API key, proper integration will have to be done separately.

### CODE REVIEW

**Comment by NahuLD** [comment]
Everything seems to be working, but I had to wait 3 months for the PR mature

**Review by bramhaag — CHANGES_REQUESTED** [review]
cya next year

File: app/src/main/java/me/bramhaag/owouploader/activity/LoginActivity.java [hunk 1]
```diff
@@ -22,47 +22,82 @@
...
+                public void onComplete(@NonNull UserModel result) {
+                    try {
+                        PreferenceManager.getDefaultSharedPreferences(LoginActivity.this).edit()
+                                .putString(
+                                    CryptographyHelper.KEY_ALIAS,
+                                    CryptographyHelper.getInstance().encrypt(inputKey)
+                                ).apply();
+                        startActivity(new Intent(LoginActivity.this, MainActivity.class));
+                    } catch (InvalidKeyException | BadPaddingException | IllegalBlockSizeException e) {
+                        finish();
```
Comment (bramhaag) [inline app/src/main/java/me/bramhaag/owouploader/activity/LoginActivity.java:97]: Maybe we should make a Toast message here explaining to the user that something unexpected happened, unless you're convinced that this should never happen

File: app/src/main/java/me/bramhaag/owouploader/activity/MainActivity.java [hunk 2]
```diff
@@ -181,6 +182,10 @@ public UploadHandler getUploadHandler() {
         return uploadHandler;
     }
 
+    public OwOAPI getApi() {
```
Comment (bramhaag) [inline app/src/main/java/me/bramhaag/owouploader/activity/MainActivity.java:185]: why? is this used ever?

File: app/src/main/java/me/bramhaag/owouploader/activity/LoginActivity.java (hunk 1 above)
Comment (NahuLD) [reply app/src/main/java/me/bramhaag/owouploader/activity/LoginActivity.java:97]: Imagine actually handling errors and not just closing the app :sunglasses:

File: app/src/main/java/me/bramhaag/owouploader/activity/MainActivity.java (hunk 2 above)
Comment (NahuLD) [reply app/src/main/java/me/bramhaag/owouploader/activity/MainActivity.java:185]: No, I thought it was needed to make DI work

Comment (bramhaag) [reply app/src/main/java/me/bramhaag/owouploader/activity/MainActivity.java:185]: it's not, you can get rid of it

**Review by bramhaag — APPROVED** [review]

**Pull request merged**